src/main/webapp/mainPage.html

```diff
@@ -0,0 +1,176 @@
+<!DOCTYPE html>
+<html lang="en">
+<head>
+    <meta charset="UTF-8">
+    <meta name="viewport" content="width=device-width, initial-scale=1.0">
+    <title>Main Page</title>
+</head>
+<style>
+    body {
+        background-color: #f2f2f2;
+    }
+    form {
+        background-color: #fff;
+        border: 1px solid #ccc;
+        border-radius: 5px;
+        box-shadow: 0 0 10px #ccc;
+        padding: 20px;
+        width: 300px;
+        margin: 0 auto;
+        margin-top: 20px;
+    }
+    form input {
+        width : 100%;
+        display: block;
+        margin: 10px 0;
+    }
+    form input[type="submit"] {
+        background-color: #4CAF50;
+        border: none;
+        color: #fff;
+        padding: 10px;
+        width: 100%;
+    }
+    form input[type="submit"]:hover {
+        cursor: pointer;
+    }
+
+    .repliedDiv{
+        background-color: #fff;
+        border: 1px solid #ccc;
+        border-radius: 5px;
+        box-shadow: 0 0 10px #ccc;
+        padding: 20px;
+        width: 300px;
+        margin: 0 auto;
+        margin-top: 20px;
+    }
+</style>
+<body onload="getPosts()">
+    <form action="/new-post" method="post">
+        <input name = "content">
+        <input name = "userId" style="display:none" id="owner_id">
+        <input type="submit" value="submit">
+    </form>
+    <hr>
+    <p>Unreplied Post:</p>
+    <div id="unRepliedWrapper">
+
+    </div>
+    <hr>
+    <p>Replied Post:</p>
+    <div id="repliedWrapper">
+
+    </div>
+
+<script>
+    //get id parameter from url
+    async function getPosts()
```

Comment:
Try to break this large function into smaller ones. I suggested one way in the other comment.